src/WebJobs.Extensions.SendGrid/Bindings/SendGridMessageAsyncCollector.cs

```diff
@@ -33,7 +34,7 @@ public Task AddAsync(SendGridMessage item, CancellationToken cancellationToken =
 
             SendGridHelpers.DefaultMessageProperties(item, _config, _attribute);
 
-            if (item.To == null || item.To.Length == 0)
+            if (item.Personalization?.Count == 0)
```

Comment:
This error message is specific to a To address - shouldn't we be checking personalization.To?